### ISSUE TITLE
Ensure deployment validity via admission hook

### ISSUE THREAD

**Issue by alpe** [substratusai/lingo#46]
Controllers can register a webhook in k8s to receive [admission callbacks](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/). This would be very useful for Lingo to ensure valid model tags and to prevent duplicate deployments for a model.

**Comment by nstogner** [comment]
I agree this would be nice to have. However, I tend to be weary of admission controllers that operate against commonly used built-in resources. If you are going to guarantee that your validating webhook is called it needs to fail closed (`failurePolicy: Fail`). This means if lingo is unresponsive for some reason you have blocked all operations against Deployments in the cluster. I lean towards punting on this feature until we have more use cases for it. In the mean time we should probably just log the failure and refuse to scale up when there is ambiguity (same model registered under multiple Deployments).
